alerts/cluster-monitoring-operator/PrometheusOperatorRejectedResources.md

```diff
@@ -0,0 +1,88 @@
+# PrometheusOperatorRejectedResources
+
+## Meaning
+
+The `PrometheusOperatorRejectedResources` alert triggers when Prometheus Operator
+rejects invalid `Alertmanager`, `PodMonitor`, `ServiceMonitor`, or `PrometheusRules`
```

Comment:
PrometheusOperatorRejectedResources details
```
        - alert: PrometheusOperatorRejectedResources
          annotations:
            description: Prometheus operator in {{ $labels.namespace }} namespace rejected
              {{ printf "%0.0f" $value }} {{ $labels.controller }}/{{ $labels.resource }}
              resources.
            summary: Resources rejected by Prometheus operator
          expr: |
            min_over_time(prometheus_operator_managed_resources{state="rejected",job="prometheus-operator", namespace=~"openshift-monitoring|openshift-user-workload-monitoring"}[5m]) > 0
          for: 5m
          labels:
            severity: warning
```
but did not see `Alertmanager` in the search, found `Probe`
```
$ token=`oc create token prometheus-k8s -n openshift-monitoring`
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?' --data-urlencode 'query=count (prometheus_operator_managed_resources{state="rejected",job="prometheus-operator"}) by (resource)' | jq
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "resource": "PodMonitor"
        },
        "value": [
          1691035010.136,
          "1"
        ]
      },
      {
        "metric": {
          "resource": "Probe"
        },
        "value": [
          1691035010.136,
          "1"
        ]
      },
      {
        "metric": {
          "resource": "PrometheusRule"
        },
        "value": [
          1691035010.136,
          "1"
        ]
      },
      {
        "metric": {
          "resource": "ServiceMonitor"
        },
        "value": [
          1691035010.136,
          "1"
        ]
      }
    ]
  }
}
```

Reply:
NNot sure why the `Alertmanager` resource is not listed, but the Alertmanager in `openshift-user-workload-monitoring` is declared in this manifest of `AlertManager` https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/alertmanager-user-workload/alertmanager.yaml  
It should generate an error if that resource is misconfigured.

Reply:
`Alertmanager` -> `AlertmanagerConfig`